diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2017-01-15 18:29:40.000000000 +0100
+++ b/CMakeLists.txt	2019-11-27 11:52:33.306219827 +0100
@@ -211,7 +211,15 @@
 endif()
 
 if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL )
-    install(TARGETS zlib zlibstatic
+    if (BUILD_SHARED_LIBS)
+        set(ZLIB_TARGETS zlib)
+        set_target_properties(zlibstatic PROPERTIES EXCLUDE_FROM_ALL ON)
+    else()
+        set(ZLIB_TARGETS zlibstatic)
+        set_target_properties(zlib PROPERTIES EXCLUDE_FROM_ALL ON)
+    endif()
+
+    install(TARGETS ${ZLIB_TARGETS}
         RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
         ARCHIVE DESTINATION "${INSTALL_LIB_DIR}"
         LIBRARY DESTINATION "${INSTALL_LIB_DIR}" )
