diff -x '.*' -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2020-02-18 21:10:25.000000000 +0100
+++ b/CMakeLists.txt	2020-02-18 21:16:21.000000000 +0100
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.15 FATAL_ERROR)
 project("Mapbox GL Native" LANGUAGES CXX C)
 
 find_program(CCACHE_PROGRAM ccache)
@@ -1022,6 +1022,20 @@
     message(FATAL_ERROR "Unsupported target platform: " ${CMAKE_SYSTEM_NAME})
 endif()
 
+if(MBGL_OPTIONAL)
 add_subdirectory(${PROJECT_SOURCE_DIR}/test)
 add_subdirectory(${PROJECT_SOURCE_DIR}/benchmark)
 add_subdirectory(${PROJECT_SOURCE_DIR}/render-test)
+endif()
+
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/mbgl TYPE INCLUDE)
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/platform/default/include/mbgl TYPE INCLUDE)
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/vendor/mapbox-base/mapbox/geometry.hpp/include/mapbox TYPE INCLUDE)
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/vendor/mapbox-base/mapbox/variant/include/mapbox TYPE INCLUDE)
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/vendor/mapbox-base/mapbox/geojson.hpp/include/mapbox TYPE INCLUDE)
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/vendor/mapbox-base/mapbox/typewrapper/include/ TYPE INCLUDE)
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/vendor/mapbox-base/mapbox/value/include/ TYPE INCLUDE)
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/vendor/mapbox-base/mapbox/weak/include/ TYPE INCLUDE)
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/vendor/mapbox-base/extras/expected-lite/include/nonstd TYPE INCLUDE)
+install(FILES ${PROJECT_SOURCE_DIR}/vendor/mapbox-base/mapbox/optional/optional.hpp TYPE INCLUDE)
+install(TARGETS mbgl-core mbgl-vendor-nunicode)
\ No newline at end of file
diff -x '.*' -Naur a/platform/default/src/mbgl/util/compression.cpp b/platform/default/src/mbgl/util/compression.cpp
--- a/platform/default/src/mbgl/util/compression.cpp	2020-02-18 21:10:29.000000000 +0100
+++ b/platform/default/src/mbgl/util/compression.cpp	2020-02-18 21:22:05.000000000 +0100
@@ -1,10 +1,6 @@
 #include <mbgl/util/compression.hpp>
 
-#if defined(__QT__) && defined(_WIN32) && !defined(__GNUC__)
-#include <QtZlib/zlib.h>
-#else
 #include <zlib.h>
-#endif
 
 #include <cstdio>
 #include <cstring>
--- a/platform/qt/qt.cmake	2020-02-18 21:10:29.000000000 +0100
+++ b/platform/qt/qt.cmake	2020-02-18 21:19:55.000000000 +0100
@@ -78,23 +78,25 @@
     PRIVATE ${PROJECT_SOURCE_DIR}/platform/default/include
 )
 
-include(${PROJECT_SOURCE_DIR}/vendor/icu.cmake)
+find_package(ZLIB REQUIRED)
+find_package(ICU COMPONENTS i18n uc REQUIRED)
 include(${PROJECT_SOURCE_DIR}/vendor/nunicode.cmake)
-include(${PROJECT_SOURCE_DIR}/vendor/sqlite.cmake)
+find_package(sqlite3 CONFIG REQUIRED)
 
 target_link_libraries(
     mbgl-core
     PRIVATE
-        $<$<NOT:$<PLATFORM_ID:Windows>>:z>
-        $<$<NOT:$<PLATFORM_ID:Windows>>:mbgl-vendor-icu>
+        ZLIB::ZLIB
+        ICU::i18n ICU::uc
         Qt5::Core
         Qt5::Gui
         Qt5::Network
         Qt5::OpenGL
         mbgl-vendor-nunicode
-        mbgl-vendor-sqlite
+        sqlite3
 )
 
+if(MBGL_OPTIONAL)
 add_library(
     qmapboxgl SHARED
     ${PROJECT_SOURCE_DIR}/platform/qt/include/qmapbox.hpp
@@ -212,3 +214,4 @@
 endif()
 
 add_test(NAME mbgl-test-runner COMMAND mbgl-test-runner WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
+endif()
\ No newline at end of file
