diff -Naur a/config-msvc.h b/config-msvc.h
--- a/config-msvc.h	2021-02-07 16:53:46.000000000 +0100
+++ b/config-msvc.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,14 +0,0 @@
-/* ./config-msvc.h.  Generated from config-msvc.h.in by configure.  */
-/* ./config-msvc.h.in - manually maintained */
-
-/* Must be =64 in order to enable huge-file support. */
-#define _FILE_OFFSET_BITS 64
-
-/* Must be defined in order to enable huge-file support. */
-#define _LARGEFILE_SOURCE 1
-
-/* Must be defined in order to enable huge-file support. */
-#define _LARGE_FILE 1
-
-/* includes gaiaconfig-msvc.h */
-#include "./src/headers/spatialite/gaiaconfig-msvc.h"
diff -Naur a/config-msvc.h.in b/config-msvc.h.in
--- a/config-msvc.h.in	2021-02-07 16:53:46.000000000 +0100
+++ b/config-msvc.h.in	2021-06-29 14:24:08.769030800 +0200
@@ -10,4 +10,4 @@
 #undef _LARGE_FILE
 
 /* includes gaiaconfig-msvc.h */
-#include "./src/headers/spatialite/gaiaconfig-msvc.h"
+#include "spatialite/gaiaconfig-msvc.h"
diff -Naur a/src/headers/spatialite/gaiaconfig.h b/src/headers/spatialite/gaiaconfig.h
--- a/src/headers/spatialite/gaiaconfig.h	2021-02-07 16:53:46.000000000 +0100
+++ b/src/headers/spatialite/gaiaconfig.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,62 +0,0 @@
-/* ./src/headers/spatialite/gaiaconfig.h.  Generated from gaiaconfig.h.in by configure.  */
-/* ./src/headers/spatialite/gaiaconfig.h.in - manually maintained */
-
-/* Should be defined in order to enable GCP support. */
-#define ENABLE_GCP 1
-
-/* Should be defined in order to enable GeoPackage support. */
-#define ENABLE_GEOPACKAGE 1
-
-/* Should be defined in order to enable LIBXML2 support. */
-#define ENABLE_LIBXML2 1
-
-/* Should be defined in order to enable MiniZIP support. */
-#define ENABLE_MINIZIP 1
-
-/* Should be defined in order to enable RTTOPO support. */
-#define ENABLE_RTTOPO 1
-
-/* Should be defined in order to enable GEOS_370 support. */
-#define GEOS_370 1
-
-/* Should be defined in order to enable GEOS_ADVANCED support. */
-#define GEOS_ADVANCED 1
-
-/* Should be defined in order to fully disable GEOS non-thread-safe API. */
-/* #undef GEOS_ONLY_REENTRANT */
-
-/* Should be defined in order to enable GEOS_REENTRANT (fully thread-safe). */
-#define GEOS_REENTRANT 1
-
-/* Should be defined in order to disable EPSG full support. */
-/* #undef OMIT_EPSG */
-
-/* Should be defined in order to disable FREEXL support. */
-/* #undef OMIT_FREEXL */
-
-/* Should be defined in order to disable GEOCALLBACKS support. */
-#define OMIT_GEOCALLBACKS 1
-
-/* Should be defined in order to disable GEOS support. */
-/* #undef OMIT_GEOS */
-
-/* Should be defined in order to disable ICONV support. */
-/* #undef OMIT_ICONV */
-
-/* Should be defined in order to disable KNN support. */
-/* #undef OMIT_KNN */
-
-/* Should be defined in order to disable MATHSQL support. */
-/* #undef OMIT_MATHSQL */
-
-/* Should be defined in order to disable PROJ.4 support. */
-/* #undef OMIT_PROJ */
-
-/* Should be defined in order to enable PROJ.6 support. */
-#define PROJ_NEW 1
-
-/* Should contain a text-string describing the intended target CPU */
-#define SPATIALITE_TARGET_CPU "x86_64-redhat-linux"
-
-/* the Version of this package */
-#define SPATIALITE_VERSION "5.0.1"
diff -Naur a/src/headers/spatialite/gaiaconfig.h.in b/src/headers/spatialite/gaiaconfig.h.in
--- a/src/headers/spatialite/gaiaconfig.h.in	2021-02-07 16:53:46.000000000 +0100
+++ b/src/headers/spatialite/gaiaconfig.h.in	2021-06-29 14:27:05.039920800 +0200
@@ -1,37 +1,37 @@
 /* ./src/headers/spatialite/gaiaconfig.h.in - manually maintained */
 
 /* Should be defined in order to enable GCP support. */
-#undef ENABLE_GCP
+#cmakedefine ENABLE_GCP 1
 
 /* Should be defined in order to enable GeoPackage support. */
-#undef ENABLE_GEOPACKAGE
+#cmakedefine ENABLE_GEOPACKAGE 1
 
 /* Should be defined in order to enable LIBXML2 support. */
-#undef ENABLE_LIBXML2
+#cmakedefine ENABLE_LIBXML2 1
 
 /* Should be defined in order to enable MiniZIP support. */
 #undef ENABLE_MINIZIP
 
 /* Should be defined in order to enable RTTOPO support. */
-#undef ENABLE_RTTOPO
+#cmakedefine ENABLE_RTTOPO 1
 
 /* Should be defined in order to enable GEOS_370 support. */
 #undef GEOS_370
 
 /* Should be defined in order to enable GEOS_ADVANCED support. */
-#undef GEOS_ADVANCED
+#cmakedefine GEOS_ADVANCED 1
 
 /* Should be defined in order to fully disable GEOS non-thread-safe API. */
 #undef GEOS_ONLY_REENTRANT
 
 /* Should be defined in order to enable GEOS_REENTRANT (fully thread-safe). */
-#undef GEOS_REENTRANT
+#define GEOS_REENTRANT 1
 
 /* Should be defined in order to disable EPSG full support. */
 #undef OMIT_EPSG
 
 /* Should be defined in order to disable FREEXL support. */
-#undef OMIT_FREEXL
+#cmakedefine OMIT_FREEXL 1
 
 /* Should be defined in order to disable GEOCALLBACKS support. */
 #undef OMIT_GEOCALLBACKS
@@ -52,10 +52,10 @@
 #undef OMIT_PROJ
 
 /* Should be defined in order to enable PROJ.6 support. */
-#undef PROJ_NEW
+#define PROJ_NEW 1
 
 /* Should contain a text-string describing the intended target CPU */
-#undef SPATIALITE_TARGET_CPU
+#define SPATIALITE_TARGET_CPU "${CMAKE_SYSTEM_PROCESSOR}"
 
 /* the Version of this package */
-#undef SPATIALITE_VERSION
+#define SPATIALITE_VERSION "${CMAKE_PROJECT_VERSION}"
diff -Naur a/src/headers/spatialite/gaiaconfig-msvc.h b/src/headers/spatialite/gaiaconfig-msvc.h
--- a/src/headers/spatialite/gaiaconfig-msvc.h	2021-02-07 16:53:46.000000000 +0100
+++ b/src/headers/spatialite/gaiaconfig-msvc.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,56 +0,0 @@
-/* ./src/headers/spatialite/gaiaconfig-msvc.h.  Generated from gaiaconfig-msvc.h.in by configure.  */
-/* ./src/headers/spatialite/gaiaconfig-msvc.h.in - manually maintained */
-
-/* Should be defined in order to enable GCP support. */
-#define ENABLE_GCP 1
-
-/* Should be defined in order to enable GeoPackage support. */
-#define ENABLE_GEOPACKAGE 1
-
-/* Should be defined in order to enable LIBXML2 support. */
-#define ENABLE_LIBXML2 1
-
-/* Should be defined in order to enable RTTOPO support. */
-#define ENABLE_RTTOPO 1
-
-/* Should be defined in order to enable GEOS_370 support. */
-#define GEOS_370 1
-
-/* Should be defined in order to enable GEOS_ADVANCED support. */
-#define GEOS_ADVANCED 1
-
-/* Should be defined in order to fully disable GEOS non-thread-safe API. */
-/* #undef GEOS_ONLY_REENTRANT */
-
-/* Should be defined in order to enable GEOS_REENTRANT (fully thread-safe). */
-#define GEOS_REENTRANT 1
-
-/* Should be defined in order to disable EPSG full support. */
-/* #undef OMIT_EPSG */
-
-/* Should be defined in order to disable FREEXL support. */
-/* #undef OMIT_FREEXL */
-
-/* Should be defined in order to disable GEOCALLBACKS support. */
-#define OMIT_GEOCALLBACKS 1
-
-/* Should be defined in order to disable GEOS support. */
-/* #undef OMIT_GEOS */
-
-/* Should be defined in order to disable ICONV support. */
-/* #undef OMIT_ICONV */
-
-/* Should be defined in order to disable KNN support. */
-/* #undef OMIT_KNN */
-
-/* Should be defined in order to disable MATHSQL support. */
-/* #undef OMIT_MATHSQL */
-
-/* Should be defined in order to disable PROJ.4 support. */
-/* #undef OMIT_PROJ */
-
-/* Should be defined in order to enable PROJ.6 support. */
-#define PROJ_NEW 1
-
-/* the Version of this package */
-#define SPATIALITE_VERSION "5.0.1"
diff -Naur a/src/headers/spatialite/gaiaconfig-msvc.h.in b/src/headers/spatialite/gaiaconfig-msvc.h.in
--- a/src/headers/spatialite/gaiaconfig-msvc.h.in	2021-02-07 16:53:46.000000000 +0100
+++ b/src/headers/spatialite/gaiaconfig-msvc.h.in	2021-06-29 14:27:12.728025600 +0200
@@ -1,22 +1,22 @@
 /* ./src/headers/spatialite/gaiaconfig-msvc.h.in - manually maintained */
 
 /* Should be defined in order to enable GCP support. */
-#undef ENABLE_GCP
+#cmakedefine ENABLE_GCP 1
 
 /* Should be defined in order to enable GeoPackage support. */
-#define ENABLE_GEOPACKAGE 1
+#cmakedefine ENABLE_GEOPACKAGE 1
 
 /* Should be defined in order to enable LIBXML2 support. */
-#define ENABLE_LIBXML2 1
+#cmakedefine ENABLE_LIBXML2 1
 
 /* Should be defined in order to enable RTTOPO support. */
-#define ENABLE_RTTOPO 1
+#cmakedefine ENABLE_RTTOPO 1
 
 /* Should be defined in order to enable GEOS_370 support. */
 #define GEOS_370 1
 
 /* Should be defined in order to enable GEOS_ADVANCED support. */
-#define GEOS_ADVANCED 1
+#cmakedefine GEOS_ADVANCED 1
 
 /* Should be defined in order to fully disable GEOS non-thread-safe API. */
 /* #undef GEOS_ONLY_REENTRANT */
@@ -28,7 +28,7 @@
 #undef OMIT_EPSG
 
 /* Should be defined in order to disable FREEXL support. */
-#undef OMIT_FREEXL
+#cmakedefine OMIT_FREEXL 1
 
 /* Should be defined in order to disable GEOCALLBACKS support. */
 #define OMIT_GEOCALLBACKS 1
@@ -52,4 +52,4 @@
 #define PROJ_NEW 1
 
 /* the Version of this package */
-#undef SPATIALITE_VERSION
+#define SPATIALITE_VERSION "${CMAKE_PROJECT_VERSION}"
