diff -Naur a/hdf5-1.12.0/CMakeFilters.cmake b/hdf5-1.12.0/CMakeFilters.cmake
--- a/hdf5-1.12.0/CMakeFilters.cmake	2020-02-29 00:29:58.000000000 +0100
+++ b/hdf5-1.12.0/CMakeFilters.cmake	2020-05-11 15:30:26.312789000 +0200
@@ -53,8 +53,8 @@
       if (NOT ZLIB_FOUND)
         find_package (ZLIB) # Legacy find
         if (ZLIB_FOUND)
-          set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${ZLIB_LIBRARIES})
-          set (LINK_COMP_SHARED_LIBS ${LINK_COMP_SHARED_LIBS} ${ZLIB_LIBRARIES})
+            set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ZLIB::ZLIB)
+            set (LINK_COMP_SHARED_LIBS ${LINK_COMP_SHARED_LIBS} ZLIB::ZLIB)
         endif ()
       endif ()
     endif ()
--- a/hdf5-1.12.0/config/cmake/hdf5-config.cmake.in	2020-02-29 00:29:58.000000000 +0100
+++ b/hdf5-1.12.0/config/cmake/hdf5-config.cmake.in	2020-05-11 15:41:15.823549000 +0200
@@ -14,6 +14,7 @@
 #-----------------------------------------------------------------------------
 @PACKAGE_INIT@
 
+include(CMakeFindDependencyMacro)
 string(TOUPPER @HDF5_PACKAGE@ HDF5_PACKAGE_NAME)
 
 set (${HDF5_PACKAGE_NAME}_VALID_COMPONENTS
@@ -73,6 +74,11 @@
   set (${HDF5_PACKAGE_NAME}_JAVA_LIBRARIES "${${HDF5_PACKAGE_NAME}_JAVA_LIBRARY}")
 endif ()
 
+if (${HDF5_PACKAGE_NAME}_ENABLE_Z_LIB_SUPPORT)
+    find_dependency(ZLIB)
+    set (${HDF5_PACKAGE_NAME}_LIBRARIES ZLIB::ZLIB)
+endif ()
+
 #-----------------------------------------------------------------------------
 # Directories
 #-----------------------------------------------------------------------------
