diff -Naur a/hdf5-1.12.0/config/cmake/hdf5-config.cmake.in b/hdf5-1.12.0/config/cmake/hdf5-config.cmake.in
--- a/hdf5-1.12.0/config/cmake/hdf5-config.cmake.in	2020-02-29 00:29:58.000000000 +0100
+++ b/hdf5-1.12.0/config/cmake/hdf5-config.cmake.in	2020-05-11 19:54:54.114039243 +0200
@@ -55,8 +55,9 @@
 # Dependencies
 #-----------------------------------------------------------------------------
 if (${HDF5_PACKAGE_NAME}_ENABLE_PARALLEL)
-  set (${HDF5_PACKAGE_NAME}_MPI_C_INCLUDE_PATH "@MPI_C_INCLUDE_DIRS@")
-  set (${HDF5_PACKAGE_NAME}_MPI_C_LIBRARIES    "@MPI_C_LIBRARIES@")
+  include(CMakeFindDependencyMacro)
+  find_dependency(MPI)
+  set (${HDF5_PACKAGE_NAME}_MPI_C_LIBRARIES    "MPI::MPI_C")
   if (${HDF5_PACKAGE_NAME}_BUILD_FORTRAN)
     set (${HDF5_PACKAGE_NAME}_MPI_Fortran_INCLUDE_PATH "@MPI_Fortran_INCLUDE_DIRS@")
     set (${HDF5_PACKAGE_NAME}_MPI_Fortran_LIBRARIES    "@MPI_Fortran_LIBRARIES@")
