diff --git a/src/vcpkg/base/messages.cpp b/src/vcpkg/base/messages.cpp
index 187eaf6..938c68f 100644
--- a/src/vcpkg/base/messages.cpp
+++ b/src/vcpkg/base/messages.cpp
@@ -165,7 +165,11 @@ namespace vcpkg::msg
         m.localized_strings.resize(m.names.size());
         m.initialized = true;
 
-        std::set<StringView, std::less<>> names_set(m.names.begin(), m.names.end());
+        std::set<StringView, std::less<>> names_set;
+        for (auto& name : m.names) {
+            names_set.emplace(name);
+        }
+        
         if (names_set.size() < m.names.size())
         {
             // This will not trigger on any correct code path, so it's fine to use a naive O(n^2)
