diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2019-03-21 14:27:35.000000000 +0100
+++ b/CMakeLists.txt	2019-04-16 14:30:21.954732000 +0200
@@ -25,6 +25,7 @@
 option(PARALLELSTL_USE_PARALLEL_POLICIES "Enable parallel policies" ON)
 set(PARALLELSTL_BACKEND "tbb" CACHE STRING "Threading backend; defaults to TBB")
 
+include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
 
 if (NOT TBB_DIR)
@@ -72,5 +73,7 @@
     ${CMAKE_CURRENT_BINARY_DIR}/ParallelSTLConfig.cmake
     @ONLY)
 
-export(TARGETS ParallelSTL NAMESPACE pstl:: FILE ParallelSTLTargets.cmake)
-export(PACKAGE ParallelSTL)
+install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/pstl DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(TARGETS ParallelSTL EXPORT pstl-targets LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/pstl)
+install(EXPORT pstl-targets FILE ParallelSTLTargets.cmake NAMESPACE pstl:: DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/parallelstl)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/ParallelSTLConfig.cmake ${CMAKE_CURRENT_BINARY_DIR}/ParallelSTLConfigVersion.cmake DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/parallelstl)
diff -Naur a/ParallelSTLConfig.cmake.in b/ParallelSTLConfig.cmake.in
--- a/ParallelSTLConfig.cmake.in	2019-03-21 14:27:35.000000000 +0100
+++ b/ParallelSTLConfig.cmake.in	2019-04-16 14:49:11.836006400 +0200
@@ -17,7 +17,7 @@
 set(PARALLELSTL_BACKEND "@PARALLELSTL_BACKEND@")
 
 if(PARALLELSTL_BACKEND STREQUAL "tbb")
-    find_dependency(TBB 2018 REQUIRED tbb)
+    find_dependency(TBB REQUIRED tbb)
 endif()
 
 include("${CMAKE_CURRENT_LIST_DIR}/ParallelSTLTargets.cmake")
