diff -Naur a/cpr/CMakeLists.txt b/cpr/CMakeLists.txt
--- a/cpr/CMakeLists.txt	2020-07-08 14:20:49.000000000 +0200
+++ b/cpr/CMakeLists.txt	2020-10-02 10:55:44.476215700 +0200
@@ -18,9 +18,13 @@
     util.cpp
     )
 
-add_library(cpr::cpr ALIAS cpr)
-
-target_link_libraries(cpr PUBLIC CURL::libcurl) # todo should be private, but first dependencys in ssl_options need to be removed
+target_include_directories(cpr
+  PUBLIC 
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
+    $<INSTALL_INTERFACE:include>
+  PRIVATE
+    $<BUILD_INTERFACE:${CURL_INCLUDE_DIRS}>
+)
 
 # Set version for shared libraries.
 set_target_properties(cpr
@@ -28,4 +32,20 @@
      VERSION ${${PROJECT_NAME}_VERSION}
      SOVERSION ${${PROJECT_NAME}_VERSION_MAJOR})
 
-install(TARGETS cpr)
+set(TARGETS_EXPORT_NAME "${PROJECT_NAME}Targets")
+install(
+    TARGETS ${CPR_LIBRARIES}
+    EXPORT ${TARGETS_EXPORT_NAME}
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
+)
+
+install(
+    EXPORT ${TARGETS_EXPORT_NAME} 
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
+)
+
+if (NOT DISABLE_INSTALL_HEADERS)
+    install(DIRECTORY ${CMAKE_SOURCE_DIR}/cpr DESTINATION include)
+endif()
