diff -Naur a/cpr/session.cpp b/cpr/session.cpp
--- a/cpr/session.cpp	2020-07-08 14:20:49.000000000 +0200
+++ b/cpr/session.cpp	2020-10-02 12:27:48.681706800 +0200
@@ -352,11 +352,15 @@
 void Session::Impl::SetSslOptions(const SslOptions& opts) {
     auto curl = curl_->handle;
     if (curl) {
-        curl_easy_setopt(curl, CURLOPT_SSLCERT, opts.cert_file.c_str());
+        if (!opts.cert_file.empty()) {
+            curl_easy_setopt(curl, CURLOPT_SSLCERT, opts.cert_file.c_str());
+        }
         if (!opts.cert_type.empty()) {
             curl_easy_setopt(curl, CURLOPT_SSLCERTTYPE, opts.cert_type.c_str());
         }
-        curl_easy_setopt(curl, CURLOPT_SSLKEY, opts.key_file.c_str());
+        if (!opts.key_file.empty()) {
+            curl_easy_setopt(curl, CURLOPT_SSLKEY, opts.key_file.c_str());
+        }
         if (!opts.key_type.empty()) {
             curl_easy_setopt(curl, CURLOPT_SSLKEYTYPE, opts.key_type.c_str());
         }
