diff -Naur -x .DS_Store a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2022-10-22 22:13:47.000000000 +0200
+++ b/CMakeLists.txt	2022-10-22 22:47:25.000000000 +0200
@@ -1101,9 +1101,11 @@
     message(FATAL_ERROR "Unsupported target platform: " ${CMAKE_SYSTEM_NAME})
 endif()
 
+if(MBGL_OPTIONAL)
 add_subdirectory(${PROJECT_SOURCE_DIR}/test)
 add_subdirectory(${PROJECT_SOURCE_DIR}/benchmark)
 add_subdirectory(${PROJECT_SOURCE_DIR}/render-test)
+endif()
 
 if(EXISTS ${PROJECT_SOURCE_DIR}/internal/internal.cmake)
     include(${PROJECT_SOURCE_DIR}/internal/internal.cmake)
     