diff -Naur -x .DS_Store a/platform/qt/qt.cmake b/platform/qt/qt.cmake
--- a/platform/qt/qt.cmake	2022-10-22 23:52:56.000000000 +0200
+++ b/platform/qt/qt.cmake	2022-10-23 00:18:01.000000000 +0200
@@ -12,6 +12,7 @@
 option(MBGL_QT_WITH_INTERNAL_SQLITE "Build MapLibre GL Qt bindings with internal sqlite" OFF)
 option(MBGL_QT_DEPLOYMENT "Autogenerate files necessary for deployment" OFF)
 
+find_package(ZLIB REQUIRED)
 find_package(QT NAMES Qt6 Qt5 COMPONENTS Core REQUIRED)
 find_package(Qt${QT_VERSION_MAJOR}
              COMPONENTS Gui
@@ -26,7 +27,7 @@
 endif()
 
 if(NOT MBGL_QT_WITH_INTERNAL_SQLITE)
-    find_package(Qt${QT_VERSION_MAJOR}Sql REQUIRED)
+    find_package(sqlite3 CONFIG REQUIRED)
 else()
     message(STATUS "Using internal sqlite")
     include(${PROJECT_SOURCE_DIR}/vendor/sqlite.cmake)
@@ -94,7 +95,7 @@
         ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/offline_database.cpp
         ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/offline_download.cpp
         ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/online_file_source.cpp
-        ${PROJECT_SOURCE_DIR}/platform/$<IF:$<BOOL:${MBGL_QT_WITH_INTERNAL_SQLITE}>,default/src/mbgl/storage/sqlite3.cpp,qt/src/mbgl/sqlite3.cpp>
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/sqlite3.cpp
         ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/compression.cpp
         ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/monotonic_timer.cpp
         $<$<BOOL:${MBGL_QT_WITH_HEADLESS}>:${PROJECT_SOURCE_DIR}/platform/qt/src/mbgl/headless_backend_qt.cpp>
@@ -142,24 +143,11 @@
         Qt${QT_VERSION_MAJOR}::Core
         Qt${QT_VERSION_MAJOR}::Gui
         Qt${QT_VERSION_MAJOR}::Network
-        $<IF:$<BOOL:${MBGL_QT_WITH_INTERNAL_SQLITE}>,mbgl-vendor-sqlite,Qt${QT_VERSION_MAJOR}::Sql>
+        $<IF:$<BOOL:${MBGL_QT_WITH_INTERNAL_SQLITE}>,mbgl-vendor-sqlite,sqlite3>
         $<$<PLATFORM_ID:Linux>:$<IF:$<BOOL:${MBGL_QT_WITH_INTERNAL_ICU}>,mbgl-vendor-icu,ICU::uc>>
         mbgl-vendor-nunicode
 )
 
-set(qmaplibregl_headers
-    ${PROJECT_SOURCE_DIR}/platform/qt/include/QMapLibreGL/QMapLibreGL
-    ${PROJECT_SOURCE_DIR}/platform/qt/include/QMapLibreGL/export.hpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/include/QMapLibreGL/map.hpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/include/QMapLibreGL/Map
-    ${PROJECT_SOURCE_DIR}/platform/qt/include/QMapLibreGL/settings.hpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/include/QMapLibreGL/Settings
-    ${PROJECT_SOURCE_DIR}/platform/qt/include/QMapLibreGL/types.hpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/include/QMapLibreGL/Types
-    ${PROJECT_SOURCE_DIR}/platform/qt/include/QMapLibreGL/utils.hpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/include/QMapLibreGL/Utils
-)
-
 if (MBGL_QT_INSIDE_PLUGIN)
     add_library(qmaplibregl OBJECT)
 elseif(MBGL_QT_STATIC)
@@ -168,29 +156,6 @@
     add_library(qmaplibregl SHARED)
 endif()
 
-target_sources(
-    qmaplibregl
-    PRIVATE
-    ${qmaplibregl_headers}
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/map.cpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/map_p.hpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/settings.cpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/types.cpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/utils.cpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/utils/map_observer.cpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/utils/map_observer.hpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/utils/map_renderer.cpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/utils/map_renderer.hpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/utils/renderer_backend.cpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/utils/renderer_backend.hpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/utils/renderer_observer.hpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/utils/scheduler.cpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/utils/scheduler.hpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/utils/conversion.hpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/utils/geojson.cpp
-    ${PROJECT_SOURCE_DIR}/platform/qt/src/utils/geojson.hpp
-)
-
 # Linux/Mac: Set framework, version and headers
 set_target_properties(
     qmaplibregl PROPERTIES
@@ -286,8 +251,8 @@
     target_link_libraries(
         qmaplibregl
         PUBLIC
-            $<$<NOT:$<BOOL:${MBGL_QT_WITH_INTERNAL_SQLITE}>>:Qt${QT_VERSION_MAJOR}::Sql>
-            $<$<NOT:$<OR:$<PLATFORM_ID:Windows>,$<PLATFORM_ID:Emscripten>>>:z>
+            $<$<NOT:$<BOOL:${MBGL_QT_WITH_INTERNAL_SQLITE}>>:sqlite3>
+            ZLIB::ZLIB
     )
 endif()
 
diff -Naur a/platform/qt/QMapLibreGLConfig.cmake.in b/platform/qt/QMapLibreGLConfig.cmake.in
--- a/platform/qt/QMapLibreGLConfig.cmake.in	2022-11-09 20:12:14.279924600 +0100
+++ b/platform/qt/QMapLibreGLConfig.cmake.in	2022-11-09 20:22:31.805028400 +0100
@@ -7,9 +7,10 @@
                 COMPONENTS Network @REQUIRED_QT_VERSION@)
 
 if(@MBGL_QT_STATIC@ AND NOT @MBGL_QT_INSIDE_PLUGIN@ AND NOT @MBGL_QT_WITH_INTERNAL_SQLITE@)
-  find_dependency(Qt@QT_VERSION_MAJOR@
-                  COMPONENTS Sql @REQUIRED_QT_VERSION@)
+  find_dependency(sqlite3 CONFIG REQUIRED)
 endif()
 
+find_dependency(ZLIB REQUIRED)
+
 include("${CMAKE_CURRENT_LIST_DIR}/QMapLibreGLTargets.cmake")
 @PACKAGE_INCLUDE_QCHTARGETS@
