diff -Naur -x .DS_Store a/platform/default/src/mbgl/util/compression.cpp b/platform/default/src/mbgl/util/compression.cpp
--- a/platform/default/src/mbgl/util/compression.cpp	2022-10-22 22:55:21.000000000 +0200
+++ b/platform/default/src/mbgl/util/compression.cpp	2022-10-22 22:59:20.000000000 +0200
@@ -1,10 +1,6 @@
 #include <mbgl/util/compression.hpp>
 
-#if defined(__QT__) && (defined(_WIN32) || defined(__EMSCRIPTEN__))
-#include <QtZlib/zlib.h>
-#else
 #include <zlib.h>
-#endif
 
 #include <cstdio>
 #include <cstring>
