diff -Naur a/include/mbgl/util/indexed_tuple.hpp b/include/mbgl/util/indexed_tuple.hpp
--- a/include/mbgl/util/indexed_tuple.hpp	2020-09-21 14:32:06.000000000 +0200
+++ b/include/mbgl/util/indexed_tuple.hpp	2020-11-24 19:06:34.504438300 +0100
@@ -30,12 +30,20 @@
 
     template <class I>
     auto& get() {
+    #ifdef _WIN32
         return std::get<TypeIndex<I, Is...>::value, Ts...>(*this);
+    #else
+        return std::get<TypeIndex<I, Is...>::value>(*this);
+    #endif
     }
 
     template <class I>
     const auto& get() const {
+    #ifdef _WIN32
         return std::get<TypeIndex<I, Is...>::value, Ts...>(*this);
+    #else
+        return std::get<TypeIndex<I, Is...>::value>(*this);
+    #endif
     }
 
     template <class... Us>
