diff -Naur a/include/lyra/parser.hpp b/include/lyra/parser.hpp
--- a/include/lyra/parser.hpp	2022-11-07 22:49:52.000000000 +0100
+++ b/include/lyra/parser.hpp	2024-02-02 11:36:48.663409500 +0100
@@ -23,6 +23,10 @@
 #include <string>
 #include <type_traits>
 
+#ifndef LYRA_HELP_TEXT_LEFT_COL_SIZE
+#define LYRA_HELP_TEXT_LEFT_COL_SIZE 26
+#endif
+
 namespace lyra {
 
 namespace detail {
@@ -189,7 +193,7 @@
 			os << get_description_text(style) << "\n";
 
 		os << "OPTIONS, ARGUMENTS:\n";
-		const std::string::size_type left_col_size = 26 - 3;
+		const std::string::size_type left_col_size = LYRA_HELP_TEXT_LEFT_COL_SIZE - 3;
 		const std::string left_pad(left_col_size, ' ');
 		for (auto const & cols : get_help_text(style))
 		{
