diff -Naur a/qtbase/cmake/3rdparty/extra-cmake-modules/find-modules/FindEGL.cmake b/qtbase/cmake/3rdparty/extra-cmake-modules/find-modules/FindEGL.cmake
--- a/qtbase/cmake/3rdparty/extra-cmake-modules/find-modules/FindEGL.cmake	2023-09-13 18:02:48.000000000 +0200
+++ b/qtbase/cmake/3rdparty/extra-cmake-modules/find-modules/FindEGL.cmake	2023-10-12 09:34:16.551179500 +0200
@@ -64,6 +64,30 @@
 include(CheckCXXSourceCompiles)
 include(CMakePushCheckState)
 
+if (WIN32)
+    find_package(unofficial-angle CONFIG)
+    if(TARGET unofficial::angle::libEGL)
+        set(EGL_FOUND TRUE)
+        if (NOT TARGET EGL::EGL)
+            get_target_property(EGL_LIB_DEB unofficial::angle::libEGL IMPORTED_LOCATION_DEBUG)
+            get_target_property(EGL_LIB_REL unofficial::angle::libEGL IMPORTED_LOCATION_RELEASE)
+            get_target_property(EGL_DEFINITIONS unofficial::angle::libEGL INTERFACE_COMPILE_DEFINITIONS)
+            get_target_property(EGL_INCLUDE_DIR unofficial::angle::libEGL INTERFACE_INCLUDE_DIRECTORIES)
+            get_target_property(EGL_LINK_LIBRARIES unofficial::angle::libEGL INTERFACE_LINK_LIBRARIES)
+            get_target_property(ANGLE_DEFINITIONS unofficial::angle::libANGLE INTERFACE_COMPILE_DEFINITIONS)
+            
+            add_library(EGL::EGL UNKNOWN IMPORTED)
+            set_target_properties(EGL::EGL PROPERTIES
+                IMPORTED_LOCATION_RELEASE "${EGL_LIB_REL}"
+                IMPORTED_LOCATION_DEBUG "${EGL_LIB_DEB}"
+                INTERFACE_LINK_LIBRARIES "${EGL_LINK_LIBRARIES}"
+                INTERFACE_COMPILE_DEFINITIONS "${EGL_DEFINITIONS};${ANGLE_DEFINITIONS}"
+                INTERFACE_INCLUDE_DIRECTORIES "${EGL_INCLUDE_DIR}"
+            )
+        endif()
+    endif()
+else ()
+
 ecm_find_package_version_check(EGL)
 
 # Use pkg-config to get the directories and then use these values
@@ -170,6 +194,7 @@
 set(EGL_LIBRARIES ${EGL_LIBRARY})
 set(EGL_INCLUDE_DIRS ${EGL_INCLUDE_DIR})
 set(EGL_VERSION_STRING ${EGL_VERSION})
+endif()
 
 include(FeatureSummary)
 set_package_properties(EGL PROPERTIES
