diff -Naur a/qtbase/src/plugins/platforms/direct2d/CMakeLists.txt b/qtbase/src/plugins/platforms/direct2d/CMakeLists.txt
--- a/qtbase/src/plugins/platforms/direct2d/CMakeLists.txt	2023-09-13 18:02:48.000000000 +0200
+++ b/qtbase/src/plugins/platforms/direct2d/CMakeLists.txt	2023-10-12 09:44:13.555636800 +0200
@@ -100,7 +100,7 @@
 ## Scopes:
 #####################################################################
 
-qt_internal_extend_target(QWindowsDirect2DIntegrationPlugin CONDITION QT_FEATURE_opengl AND NOT QT_FEATURE_dynamicgl
+qt_internal_extend_target(QWindowsDirect2DIntegrationPlugin CONDITION QT_FEATURE_opengl AND NOT QT_FEATURE_dynamicgl AND NOT QT_FEATURE_opengles2
     LIBRARIES
         opengl32
 )
@@ -112,7 +112,7 @@
         ../windows/qwindowspointerhandler.cpp
 )
 
-qt_internal_extend_target(QWindowsDirect2DIntegrationPlugin CONDITION QT_FEATURE_opengl
+qt_internal_extend_target(QWindowsDirect2DIntegrationPlugin CONDITION QT_FEATURE_opengl AND QT_FEATURE_opengl_desktop
     SOURCES
         ../windows/qwindowsglcontext.cpp ../windows/qwindowsglcontext.h
         ../windows/qwindowsopenglcontext.h
