diff -Naur a/qtbase/src/opengl/CMakeLists.txt b/qtbase/src/opengl/CMakeLists.txt
--- a/qtbase/src/opengl/CMakeLists.txt	2023-09-13 18:02:48.000000000 +0200
+++ b/qtbase/src/opengl/CMakeLists.txt	2023-10-12 09:42:08.382539700 +0200
@@ -108,6 +108,8 @@
     SOURCES
         qopenglcompositor.cpp qopenglcompositor_p.h
         qopenglcompositorbackingstore.cpp qopenglcompositorbackingstore_p.h
+    LIBRARIES
+        EGL::EGL
 )
 
 qt_internal_generate_tracepoints(OpenGL opengl
