diff -Naur a/qtbase/src/gui/CMakeLists.txt b/qtbase/src/gui/CMakeLists.txt
--- a/qtbase/src/gui/CMakeLists.txt	2023-09-13 18:02:48.000000000 +0200
+++ b/qtbase/src/gui/CMakeLists.txt	2023-10-12 09:36:51.514139000 +0200
@@ -332,8 +332,13 @@
 # available until the qt_internal_add_module call.
 if(QT_FEATURE_opengl)
     if(QT_FEATURE_opengles2)
-        find_package(GLESv2)
-        target_link_libraries(Gui PUBLIC GLESv2::GLESv2)
+        if(WIN32)
+            find_package(unofficial-angle CONFIG REQUIRED)
+            target_link_libraries(Gui PUBLIC unofficial::angle::libEGL unofficial::angle::libGLESv2)
+        else()
+            find_package(GLESv2)
+            target_link_libraries(Gui PUBLIC GLESv2::GLESv2)
+        endif()
 
         if(INTEGRITY AND _qt_igy_gui_libs)
             find_package(IntegrityPlatformGraphics)
