diff -Naur a/qtbase/src/plugins/platforms/windows/CMakeLists.txt b/qtbase/src/plugins/platforms/windows/CMakeLists.txt
--- a/qtbase/src/plugins/platforms/windows/CMakeLists.txt	2023-09-13 18:02:48.000000000 +0200
+++ b/qtbase/src/plugins/platforms/windows/CMakeLists.txt	2023-10-12 09:46:35.692566900 +0200
@@ -87,7 +87,7 @@
 ## Scopes:
 #####################################################################
 
-qt_internal_extend_target(QWindowsIntegrationPlugin CONDITION QT_FEATURE_opengl
+qt_internal_extend_target(QWindowsIntegrationPlugin CONDITION QT_FEATURE_opengl AND NOT QT_FEATURE_opengles2
     SOURCES
         qwindowsglcontext.cpp qwindowsglcontext.h
         qwindowsopenglcontext.h
@@ -95,7 +95,15 @@
         Qt::OpenGLPrivate
 )
 
-qt_internal_extend_target(QWindowsIntegrationPlugin CONDITION QT_FEATURE_opengl AND NOT QT_FEATURE_dynamicgl
+qt_internal_extend_target(QWindowsIntegrationPlugin CONDITION QT_FEATURE_opengl AND QT_FEATURE_opengles2
+    SOURCES
+        qwindowseglcontext.cpp qwindowseglcontext.h
+    LIBRARIES
+        Qt::OpenGLPrivate
+        EGL::EGL
+)
+
+qt_internal_extend_target(QWindowsIntegrationPlugin CONDITION QT_FEATURE_opengl AND NOT QT_FEATURE_dynamicgl AND NOT QT_FEATURE_opengles2
     LIBRARIES
         opengl32
 )
