diff -Naur a/qtbase/cmake/QtBuild.cmake b/qtbase/cmake/QtBuild.cmake
--- a/qtbase/cmake/QtBuild.cmake	2021-10-01 13:17:40.133734200 +0200
+++ b/qtbase/cmake/QtBuild.cmake	2021-10-04 11:33:39.878203400 +0200
@@ -198,7 +198,7 @@
     if(QT_CONFIG_INSTALL_DIR)
         string(APPEND QT_CONFIG_INSTALL_DIR "/")
     endif()
-    string(APPEND QT_CONFIG_INSTALL_DIR ${__config_path_part})
+    string(APPEND QT_CONFIG_INSTALL_DIR "share")
 
     set(QT_BUILD_DIR "${QT_BUILD_DIR}" PARENT_SCOPE)
     set(QT_INSTALL_DIR "${QT_INSTALL_DIR}" PARENT_SCOPE)
