diff -Naur a/libgeotiff/cmake/project-config.cmake.in b/libgeotiff/cmake/project-config.cmake.in
--- a/libgeotiff/cmake/project-config.cmake.in	2022-10-15 17:17:37.000000000 +0200
+++ b/libgeotiff/cmake/project-config.cmake.in	2022-10-15 17:24:33.000000000 +0200
@@ -24,7 +24,7 @@
 
 include(CMakeFindDependencyMacro)
 @CONFIG_PUBLIC_DEPENDENCIES@
-set (@PROJECT_NAME@_LIBRARIES @GEOTIFF_LIBRARY_TARGET@)
+set (@PROJECT_NAME@_LIBRARIES @GEOTIFF_LIB_NAME@)
 if("@BUILD_SHARED_LIBS@")
   set (@PROJECT_NAME@_SHARED_LIBRARIES ${@PROJECT_NAME@_LIBRARIES})
 else()
