diff -Naur a/port/cpl_safemaths.hpp b/port/cpl_safemaths.hpp
--- a/port/cpl_safemaths.hpp	2023-03-07 18:14:55.000000000 +0100
+++ b/port/cpl_safemaths.hpp	2023-03-27 20:10:51.427327400 +0200
@@ -126,7 +126,7 @@
 inline CPLSafeInt<unsigned> CPLSM_TO_UNSIGNED(GUInt64 x);
 #endif
 
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && !defined(__INTEL_LLVM_COMPILER)
 class CPLMSVCSafeIntException : public msl::utilities::SafeIntException
 {
   public:
