diff -Naur a/frmts/netcdf/netcdfdataset.cpp b/frmts/netcdf/netcdfdataset.cpp
--- a/frmts/netcdf/netcdfdataset.cpp	2020-01-08 12:46:32.000000000 +0100
+++ b/frmts/netcdf/netcdfdataset.cpp	2020-07-14 09:31:11.388867900 +0200
@@ -5819,7 +5819,7 @@
         CPLFree(pszTemp);
     }
 #endif
-    status = nc_open(osFilename, NC_WRITE, &cdfid);
+    status = nc_open(osFilenameForNCOpen, NC_WRITE, &cdfid);
     NCDF_ERR(status);
     if( status != NC_NOERR )
         return false;
