diff -Naur a/cmake/FindCMath.cmake b/cmake/FindCMath.cmake
--- a/cmake/FindCMath.cmake	2021-03-10 11:56:56.000000000 +0100
+++ b/cmake/FindCMath.cmake	2021-12-16 15:27:12.099120100 +0100
@@ -32,10 +32,8 @@
 
 check_symbol_exists(pow "math.h" CMath_HAVE_LIBC_POW)
 if(NOT CMath_HAVE_LIBC_POW)
-    find_library(CMath_LIBRARY NAMES m)
-
     set(CMAKE_REQUIRED_LIBRARIES_SAVE ${CMAKE_REQUIRED_LIBRARIES})
-    set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${CMath_LIBRARY})
+    set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} "m")
     check_symbol_exists(pow "math.h" CMath_HAVE_LIBM_POW)
     set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES_SAVE})
 endif()
