diff -Naur a/qwtconfig.pri b/qwtconfig.pri
--- a/qwtconfig.pri	2019-01-02 17:22:31.000000000 +0100
+++ b/qwtconfig.pri	2020-02-28 11:05:01.014912100 +0100
@@ -72,7 +72,9 @@
 # it will be a static library.
 ######################################################################
 
+CONFIG(dynamic, dynamic|static) {
 QWT_CONFIG           += QwtDll
+}
 
 ######################################################################
 # QwtPlot enables all classes, that are needed to use the QwtPlot
