diff -Naur a/libhdf5/hdf5debug.c b/libhdf5/hdf5debug.c
--- a/libhdf5/hdf5debug.c	2020-03-27 16:33:36.000000000 +0100
+++ b/libhdf5/hdf5debug.c	2020-05-20 15:35:52.370960000 +0200
@@ -5,7 +5,7 @@
 #include "config.h"
 #include <stdarg.h>
 #include <stdio.h>
-#if !defined _WIN32 && !defined __CYGWIN__
+#ifdef _GLIBC_
 #include <execinfo.h>
 #endif
 
@@ -20,7 +20,7 @@
 int
 nch5breakpoint(int err)
 {
-#if !defined _WIN32 && !defined __CYGWIN__
+#ifdef _GLIBC_
     int count = 0;
     char** trace = NULL;
     int i;
