diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2020-03-27 16:33:36.000000000 +0100
+++ b/CMakeLists.txt	2020-05-13 10:32:44.717988000 +0200
@@ -866,7 +866,7 @@
 
 # Check for the math library so it can be explicitly linked.
 IF(NOT WIN32)
-  FIND_LIBRARY(HAVE_LIBM NAMES math m libm)
+  #FIND_LIBRARY(HAVE_LIBM NAMES math m libm)
   MESSAGE(STATUS "Found Math library: ${HAVE_LIBM}")
   IF(NOT HAVE_LIBM)
     MESSAGE(FATAL_ERROR "Unable to find the math library.")
@@ -1875,11 +1875,15 @@
 SET(NC_LIBS "")
 
 FOREACH(_LIB ${ALL_TLL_LIBS})
+  if (${_LIB} MATCHES "^-l*")
+    LIST(APPEND NC_LIBS "${_LIB}")
+  else ()
   GET_FILENAME_COMPONENT(_LIB_NAME ${_LIB} NAME_WE)
   STRING(REGEX REPLACE "^lib" "" _NAME ${_LIB_NAME})
   LIST(APPEND NC_LIBS "-l${_NAME}")
   GET_FILENAME_COMPONENT(_LIB_DIR ${_LIB} PATH)
   LIST(APPEND LINKFLAGS "-L${_LIB_DIR}")
+  endif()
 ENDFOREACH()
 
 #SET(NC_LIBS "-lnetcdf ${NC_LIBS}")
