diff -Naur a/cmake/scripts/generate_config.cmake b/cmake/scripts/generate_config.cmake
--- a/cmake/scripts/generate_config.cmake	2023-09-19 15:00:22.000000000 +0200
+++ b/cmake/scripts/generate_config.cmake	2024-01-16 14:02:07.351125900 +0100
@@ -1,4 +1,4 @@
-##===----------------------------------------------------------------------===##
+# #===----------------------------------------------------------------------===##
 #
 # Copyright (C) Intel Corporation
 #
@@ -10,13 +10,13 @@
 # Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
 # See https://llvm.org/LICENSE.txt for license information.
 #
-##===----------------------------------------------------------------------===##
+# #===----------------------------------------------------------------------===##
 
-if (NOT OUTPUT_DIR)
+if(NOT OUTPUT_DIR)
     set(OUTPUT_DIR "output")
 endif()
 
-if (SKIP_HEADERS_SUBDIR)
+if(SKIP_HEADERS_SUBDIR)
     set(HANDLE_HEADERS_PATH "
 get_filename_component(_onedpl_headers \"\${_onedpl_root}/include\" ABSOLUTE)
 ")
@@ -28,7 +28,7 @@
     set(_onedpl_headers_subdir linux)
 endif()
 
-get_filename_component(_onedpl_headers \"\${_onedpl_root}/\${_onedpl_headers_subdir}/include\" ABSOLUTE)
+get_filename_component(_onedpl_headers \"\${_onedpl_root}/include\" ABSOLUTE)
 ")
 endif()
 
@@ -42,13 +42,13 @@
 set(PROJECT_VERSION "${_onedpl_ver_major}.${_onedpl_ver_minor}.${_onedpl_ver_patch}")
 
 configure_file("${ONEDPL_ROOT}/cmake/templates/oneDPLConfig.cmake.in"
-               "${OUTPUT_DIR}/oneDPLConfig.cmake"
-               @ONLY)
+    "${OUTPUT_DIR}/oneDPLConfig.cmake"
+    @ONLY)
 configure_file("${ONEDPL_ROOT}/cmake/templates/oneDPLConfigVersion.cmake.in"
-               "${OUTPUT_DIR}/oneDPLConfigVersion.cmake"
-               @ONLY)
+    "${OUTPUT_DIR}/oneDPLConfigVersion.cmake"
+    @ONLY)
 
-if (SKIP_HEADERS_SUBDIR)
+if(SKIP_HEADERS_SUBDIR)
     set(_onedpl_pkgconfig_header_suffix include)
     configure_file("${ONEDPL_ROOT}/integration/pkgconfig/dpl.pc.in" "${OUTPUT_DIR}/dpl.pc" @ONLY)
 else()
diff -Naur a/cmake/templates/oneDPLConfig.cmake.in b/cmake/templates/oneDPLConfig.cmake.in
--- a/cmake/templates/oneDPLConfig.cmake.in	2023-09-19 15:00:22.000000000 +0200
+++ b/cmake/templates/oneDPLConfig.cmake.in	2024-01-16 13:25:43.983164800 +0100
@@ -32,7 +32,7 @@
 
 # Installation path: <onedpl_root>/lib/cmake/oneDPL/
 get_filename_component(_onedpl_root "${CMAKE_CURRENT_LIST_DIR}" REALPATH)
-get_filename_component(_onedpl_root "${_onedpl_root}/../../../" ABSOLUTE)
+get_filename_component(_onedpl_root "${_onedpl_root}/../../" ABSOLUTE)
 @HANDLE_HEADERS_PATH@
 if (EXISTS "${_onedpl_headers}")
     if (NOT TARGET oneDPL)
