diff -Naur a/dev/restinio/CMakeLists.txt b/dev/restinio/CMakeLists.txt
--- a/dev/restinio/CMakeLists.txt	2021-11-19 14:31:38.000000000 +0100
+++ b/dev/restinio/CMakeLists.txt	2022-05-19 09:58:21.480109100 +0200
@@ -53,7 +53,7 @@
 	IF (RESTINIO_FMT_HEADER_ONLY)
 		TARGET_LINK_LIBRARIES(${RESTINIO} INTERFACE fmt::fmt-header-only)
 	ELSE ()
-		TARGET_LINK_LIBRARIES(${RESTINIO} PUBLIC fmt::fmt)
+		TARGET_LINK_LIBRARIES(${RESTINIO} INTERFACE fmt::fmt)
 	ENDIF ()
 
 	IF (RESTINIO_USE_EXTERNAL_HTTP_PARSER)
