diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2022-10-17 09:10:02.000000000 +0200
+++ b/CMakeLists.txt	2022-11-21 17:06:51.103493500 +0100
@@ -51,6 +51,7 @@
 )
 
 include(PCRaster)
+include_directories(${Boost_INCLUDE_DIR})
 
 add_subdirectory(source)
 add_subdirectory(documentation)
diff -Naur a/environment/cmake/PCRasterConfiguration.cmake b/environment/cmake/PCRasterConfiguration.cmake
--- a/environment/cmake/PCRasterConfiguration.cmake	2022-10-17 09:10:02.000000000 +0200
+++ b/environment/cmake/PCRasterConfiguration.cmake	2022-11-21 16:56:34.572428300 +0100
@@ -162,18 +162,16 @@
 find_package(XercesC REQUIRED)
 
 
-find_package(GDAL 2.4 REQUIRED)
+find_package(Gdal CONFIG REQUIRED)
 message(STATUS "Found GDAL: ")
-message(STATUS "  version:   " ${GDAL_VERSION})
-message(STATUS "  libraries: " ${GDAL_LIBRARIES})
-message(STATUS "  includes:  " ${GDAL_INCLUDE_DIRS})
+message(STATUS "  version:   " ${Gdal_VERSION})
+message(STATUS "  libraries: " ${Gdal_LIBRARIES})
+message(STATUS "  includes:  " ${Gdal_INCLUDE_DIRS})
 find_program(GDAL_TRANSLATE gdal_translate
     HINTS ${GDAL_INCLUDE_DIRS}/../bin
 )
-if(EXISTS "${GDAL_INCLUDE_DIRS}/../../share/gdal")
-    set(GDAL_DATA "${GDAL_INCLUDE_DIRS}/../../share/gdal")
-elseif(EXISTS "${GDAL_INCLUDE_DIRS}/../share/gdal")
-    set(GDAL_DATA "${GDAL_INCLUDE_DIRS}/../share/gdal")
+if(Gdal_DATA_PATH)
+    set(GDAL_DATA "${Gdal_DATA_PATH}")
 else()
     message(FATAL_ERROR "GDAL data dir not found")
 endif()
