diff -Naur a/src/libGLESv2/global_state.cpp b/src/libGLESv2/global_state.cpp
--- a/src/libGLESv2/global_state.cpp	2020-05-15 12:44:40.000000000 +0200
+++ b/src/libGLESv2/global_state.cpp	2021-08-02 15:17:32.169941700 +0200
@@ -262,6 +262,7 @@
 #    endif  // defined(ANGLE_ENABLE_ASSERTS) && !defined(ANGLE_ENABLE_WINDOWS_UWP)
 }  // namespace
 
+#ifndef ANGLE_STATIC
 extern "C" BOOL WINAPI DllMain(HINSTANCE instance, DWORD reason, LPVOID)
 {
     switch (reason)
@@ -285,4 +286,5 @@
 
     return TRUE;
 }
+#endif
 #endif  // ANGLE_PLATFORM_WINDOWS
